index.js

```diff
@@ -50,9 +71,31 @@ getUsersCreditDetails(usersArray);
 
 const genderView = users => {
   // Your code goes here ...
+  const femaleUsers = [];
+  const maleUsers = [];
+
+let females = users.filter((female) => {
+	return female.gender === "female";
+});
+
+for(let {firstName, lastName} of females){
+  femaleUsers.push(`${firstName} ${lastName}`)
+}
+
+let males = users.filter((male) => {
+	return male.gender === "male";
+});
+
+for(let {firstName, lastName} of males){
+  maleUsers.push(`${firstName} ${lastName}`)
+}
```

Comment:
even tough this is totally correct, you could have saved yourself multiple lines of code.

You have already implemented a  function that returns the full names of users.

After the filter you could have just returned:
```js
return {
    femaleUsers: getFullNames(females),
    maleUsers: getFullNames(males)
}
```

DRY code is not just a fancy way of telling people to write less code, but it also saves keystrokes and eventually possible errors

Reply:
So doing it this way I'm using the previously defined functions instead of writing everything individually?

Reply:
yes, exactly. not repeating logic youve already done

Reply:
const genderView = users => {
  // Your code goes here ...
let females = users.filter((female) => {
	return female.gender === "female";
});

let males = users.filter((male) => {
	return male.gender === "male";
});

return {femaleUsers: getFullNames(males), maleUsers: getFullNames(females)}
};

genderView(usersArray)

Way better!  Thanks!